components/datetime/src/provider/date_time.rs

```diff
@@ -8,102 +8,141 @@ use crate::fields;
 use crate::options::{components, length, preferences, DateTimeFormatOptions};
 use crate::pattern::{hour_cycle, reference::Pattern};
 use crate::provider;
+use crate::provider::gregory::{
+    DatePatternsV1Marker, DateSkeletonPatternsV1Marker, DateSymbolsV1Marker,
+};
 use crate::skeleton;
 use alloc::borrow::Cow;
+use icu_locid::Locale;
+use icu_provider::prelude::*;
 
 type Result<T> = core::result::Result<T, DateTimeFormatError>;
 
-pub trait DateTimePatterns {
-    fn get_pattern_for_options(&self, options: &DateTimeFormatOptions) -> Result<Option<Pattern>>;
-    fn get_pattern_for_components_bag(
-        &self,
-        components: &components::Bag,
-    ) -> Result<Option<Pattern>>;
-    fn get_pattern_for_length_bag(&self, length: &length::Bag) -> Result<Option<Pattern>>;
-    fn get_pattern_for_date_length(&self, length: length::Date) -> Result<Pattern>;
-    fn get_pattern_for_time_length(
-        &self,
-        length: length::Time,
-        preferences: &Option<preferences::Bag>,
-    ) -> Result<Pattern>;
-    fn get_pattern_for_datetime_length(
-        &self,
-        length: length::Date,
-        date: Pattern,
-        time: Pattern,
-    ) -> Result<Pattern>;
+pub(crate) fn pattern_for_options<'data, D>(
```

Comment:
Thought: I miss the verbs on these methods, as it gives a hint to what is going on under the hood. It wasn't obvious to me that this is part of the codepath that invokes the components bag machinery.

Reply:
If you're referring to `get_pattern_for_options` vs `pattern_for_options` it's a Rust naming scheme that I follow - https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter .